src/firefly/java/edu/caltech/ipac/astro/FITSTableReader.java

```diff
@@ -210,53 +209,43 @@ else if (strategy == FULLY_FLATTEN) {
     /**
      * Get a list of star tables from a FITS file.
      * @param fits_filename
+     * @param table_idx
      * @return List of star tables.
      * @throws IOException
      */
-    public static List getStarTableList(String fits_filename) {
 
-        ArrayList<StarTable> tList = new ArrayList<>();
+    public static List getStarTableList(String fits_filename, int table_idx) {
```

Comment:
I understand the need for this change. But, it should be a new method.  i.e.
`public static StarTable getStarTable(String fits_filename, int table_idx) `

If the original method is not used and cannot be easily implemented, I would just throw an unsupported exception for now.